### PR TITLE
Rename "at-rules-without-declaration-blocks" to "blockless-at-rules" in max-nesting-depth

### LINKED ISSUE #2231: Rename ignore: ["at-rules-without-declaration-blocks"] to ["blockless-at-rules"] in max-nesting-depth
https://github.com/stylelint/stylelint/issues/2229 makes me think it's worth being very explicit in our option names. Here we have a rule that targets (nested) rules and might have options that target rules _and at-rules_.

In https://github.com/stylelint/stylelint/pull/2225  I think we're being a bit implicit with the naming of the option `"after-same-name"`.  I think we should be more explicit i.e.:

- `"after-same-name"` becomes `"after-same-name-at-rule"`

In the [deprecations branch](https://github.com/stylelint/stylelint/tree/deprecations/lib/rules/at-rule-empty-line-before#except-inside-block-blockless-after-same-name-blockless-blockless-after-blockless-after-opening-brace) we have `blockless-after-same-name-blockless"` and `"blockless-after-blockless"`. I think we should be more explicit i.e.:

- `"blockless-after-blockless"` becomes `"blockless-at-rule-after-blockless-at-rule"`
- `blockless-after-same-name-blockless"` becomes `blockless-at-rule-after-same-name-blockless-at-rule"`

It's a bit more verbose, but it means that constructs are consistently named across rules and their options e.g. `rule-nested-empty-line-before` has the option `"after-rule"` and, if added, will have the option `"after-at-rule"`.

Lastly, there's an opportunity to standardise on the "blockless-at-rule" name in `max-nesting-depth`:

- [`ignore: ["at-rules-without-declaration-blocks"]`](https://github.com/stylelint/stylelint/blob/master/lib/rules/max-nesting-depth/README.md#ignore-at-rules-without-declaration-blocks) becomes `ignore: ["blockless-at-rules"]`.

Thoughts?

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

Closes #2231.

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/max-nesting-depth/README.md [hunk 1]
```diff
@@ -96,7 +96,9 @@ a .foo__foo .bar .baz {}
 
 ## Optional secondary options
 
-### `ignore: ["at-rules-without-declaration-blocks"]`
+### `ignore: ["blockless-at-rules"]`
+
+***Note: This option was previously called `at-rules-without-declaration-blocks`. See [the release planning docs](http://stylelint.io/user-guide/release-planning/) for details.***
```
Comment (jeddy3) [inline lib/rules/max-nesting-depth/README.md]: Let's drop the "See [the release planning docs](http://stylelint.io/user-guide/release-planning/) for details." bit.

I made a mistake advocating that in another PR. The releasing planning doc is a call for community help, and no help is required with these option renames.

**Comment by davidtheclark** [comment]
Thanks! @jeddy3 @hudochenkov How are we handling changelog entries for the deprecations branch?

**Comment by hudochenkov** [comment]
@davidtheclark @jeddy3 I can't find any changelog messages for deprecated rules and options.

Nevertheless there is an issue with a list of deprecated rules https://github.com/stylelint/stylelint/issues/2123 and PR with some of these rules deprecated https://github.com/stylelint/stylelint/pull/2197.

List of deprecated options https://github.com/stylelint/stylelint/issues/1663#issuecomment-267795609 (subject to change), PR with deprecation https://github.com/stylelint/stylelint/pull/2213.

And current PR https://github.com/stylelint/stylelint/pull/2275.

Shall we add everything deprecated to changelog in `deprecations` branch?

**Comment by jeddy3** [comment]
> Shall we add everything deprecated to changelog in deprecations branch?

Yes please!

They've historically been written as:

> - Deprecated: `number-zero-length-no-unit`. Use `length-zero-no-unit` instead.
> - Deprecated: `"single-where-required"` option for `font-family-name-quotes`. Use the `"always-unless-keyword"` option instead.
> - Deprecated: `x`. If you use this option/rule, please consider creating a plugin for the community. See the [release planning document](docs/user-guide/release-planning.md) for more details.

FYI, I forgot to mention that the changelog is written in the following order:

- Removed
- Changed
- Deprecated
- Added
- Fixed

And within each group it is order by general items first (i.e. the engine, CLI etc), follow by rule specific items (order alphabetically by the rule name e.g. "Fixed: `indentation` no longer warns ...")

**Comment by hudochenkov** [comment]
@jeddy3 I'm not sure where to put these changelog records, because [changelog](https://github.com/stylelint/stylelint/blob/deprecations/CHANGELOG.md) in `deprecataions` branch is a little outdated.

---

- Deprecated: `block-no-single-line`. Use `block-opening-brace-newline-after` and `block-closing-brace-newline-before` rules with the option `"always"` instead. See [the FAQs for an example](docs/user-guide/faq.md#how-do-i-disallow-single-line-blocks).
- Deprecated: `declaration-block-no-ignored-properties`. If you use this option/rule, please consider creating a plugin for the community.
- Deprecated: `declaration-block-properties-order`. Use the community [`stylelint-order`](https://github.com/hudochenkov/stylelint-order) plugin pack instead. See the [release planning](docs/user-guide/release-planning.md) document for more details.
- Deprecated: `media-feature-no-missing-punctuation`. If you use this option/rule, please consider creating a plugin for the community.
- Deprecated: `selector-no-empty`. If you use this option/rule, please consider creating a plugin for the community.
- Deprecated: `"all-nested"` option for `at-rule-empty-line-before`. Use the `"inside-block"` option instead.
- Deprecated: `"blockless-group"` option for `at-rule-empty-line-before`. Use the `"blockless-after-blockless"` option instead.
- Deprecated: `"between-comments"` option for `comment-empty-line-before`. Use the `"after-comment"` option instead.
- Deprecated: `"stylelint-commands"` option for `comment-empty-line-before`. Use the `"stylelint-command"` option instead.
- Deprecated: `"at-rules-without-declaration-blocks"` option for `max-nesting-depth`. Use the `"blockless-at-rules"` option instead.

**Comment by jeddy3** [comment]
Probably rebase `deprecations` off `master`, then add a new "Deprecations" heading at the top of the CHANGELOG and put them in there?

Your items LGTM. I've made a couple of tweaks (edit my post to access the raw markdown):

---

- Deprecated: `block-no-single-line`. Use `block-opening-brace-newline-after` and `block-closing-brace-newline-before` rules with the option `"always"` instead. See [the FAQs for an example](docs/user-guide/faq.md#how-do-i-disallow-single-line-blocks).
- Deprecated: `declaration-block-no-ignored-properties`. If you use this rule, please consider creating a plugin for the community. See the [release planning](docs/user-guide/release-planning.md) document for more details.
- Deprecated: `declaration-block-properties-order`. Use the community [`stylelint-order`](https://github.com/hudochenkov/stylelint-order) plugin pack instead. See the [release planning](docs/user-guide/release-planning.md) document for more details.
- Deprecated: `media-feature-no-missing-punctuation`. If you use this rule, please consider creating a plugin for the community. See the [release planning](docs/user-guide/release-planning.md) document for more details.
- Deprecated: `selector-no-empty`. If you use this rule, please consider creating a plugin for the community. See the [release planning](docs/user-guide/release-planning.md) document for more details.
- Deprecated: `"all-nested"` option for `at-rule-empty-line-before`. Use the `"inside-block"` option instead.
- Deprecated: `"blockless-group"` option for `at-rule-empty-line-before`. Use the `"blockless-after-blockless"` option instead.
- Deprecated: `"between-comments"` option for `comment-empty-line-before`. Use the `"after-comment"` option instead.
- Deprecated: `"stylelint-commands"` option for `comment-empty-line-before`. Use the `"stylelint-command"` option instead.
- Deprecated: `"at-rules-without-declaration-blocks"` option for `max-nesting-depth`. Use the `"blockless-at-rules"` option instead.

**Comment by davidtheclark** [comment]
Thanks @hudochenkov!

**Comment by hudochenkov** [comment]
[Done](https://github.com/stylelint/stylelint/blob/deprecations/CHANGELOG.md). But I forgot about “Deprecations” header :man_shrugging:

**Comment by ntwb** [comment]
> Done. But I forgot about “Deprecations” header 🤷‍♂️

I think leaving that as `HEAD` is fine...

Could also change it to `HEAD (DEPRECATION)` and `HEAD (V8)` for the two major branches we're currently juggling?

**Pull request merged**